client/rpcserver/rpcserver.go

```diff
@@ -359,12 +360,32 @@ func (s *RPCServer) handleRequest(req *msgjson.Message) *msgjson.ResponsePayload
 	// Find the correct handler for this route.
 	h, exists := routes[req.Route]
 	if !exists {
-		log.Debugf("%v: %v", ErrUnknownCmd, req.Route)
-		payload.Error = msgjson.NewError(msgjson.RPCUnknownRoute, ErrUnknownCmd.Error())
+		log.Debugf("%v: %v", errUnknownCmd, req.Route)
+		payload.Error = msgjson.NewError(msgjson.RPCUnknownRoute, errUnknownCmd.Error())
 		return payload
 	}
 
-	return h(s, req)
+	params := new(RawParams)
+	err := req.Unmarshal(params)
+	if err != nil {
+		log.Debugf("cannot unmarshal params for route %s", req.Route)
+		payload.Error = msgjson.NewError(msgjson.RPCParseError, "unable to unmarshal request")
+		return payload
+	}
+
+	args, err := parseCmdArgs(req.Route, params.PWArgs, params.Args)
+	if err != nil {
+		if errors.Is(err, errArgs) {
+			// This is a known command. Ignoring unreachable error.
+			usage, _ := commandUsage(req.Route, false)
+			err = fmt.Errorf("%v\n\n%s", err, usage)
+		}
+		log.Tracef("bad args for route %s", req.Route)
+		payload.Error = msgjson.NewError(msgjson.RPCParseError, err.Error())
+		return payload
+	}
+
+	return h(s, args)
```

Comment:
It's a little roundabout for the parsers to return `interface{}`, only for the handlers to do a type assertion. Seems like you should have the handlers call the parsers directly and skip the `interface{}` middleman.

Reply:
you are right. I'm on it.